x-pack/metricbeat/module/meraki/device_health/switchports.go

```diff
@@ -0,0 +1,155 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package device_health
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+	meraki "github.com/meraki/dashboard-api-go/v3/sdk"
+)
+
+type switchport struct {
+	port       *meraki.ResponseItemSwitchGetOrganizationSwitchPortsBySwitchPorts
+	portStatus *meraki.ResponseItemSwitchGetDeviceSwitchPortsStatuses
+}
+
+func getDeviceSwitchports(client *meraki.Client, organizationID string, devices map[Serial]*Device, period time.Duration) error {
+	switches, res, err := client.Switch.GetOrganizationSwitchPortsBySwitch(organizationID, &meraki.GetOrganizationSwitchPortsBySwitchQueryParams{})
+	if err != nil {
+		return fmt.Errorf("GetOrganizationSwitchPortsBySwitch failed; [%d] %s. %w", res.StatusCode(), res.Body(), err)
+	}
+
+	for _, device := range *switches {
+		if device.Ports == nil {
+			continue
+		}
+
+		var switchports []*switchport
+		for _, port := range *device.Ports {
+			switchports = append(switchports, &switchport{port: &port})
+		}
+
+		statuses, res, err := client.Switch.GetDeviceSwitchPortsStatuses(device.Serial, &meraki.GetDeviceSwitchPortsStatusesQueryParams{
+			Timespan: period.Seconds(),
+		})
+		if err != nil {
+			return fmt.Errorf("GetDeviceSwitchPortsStatuses failed; [%d] %s. %w", res.StatusCode(), res.Body(), err)
+		}
+
+		// match status to the port attributes found earlier using the shared port ID
+		for _, status := range *statuses {
+			for _, switchport := range switchports {
+				if switchport.port.PortID == status.PortID {
+					switchport.portStatus = &status
+					break
+				}
+			}
+		}
+
+		devices[Serial(device.Serial)].switchports = switchports
+	}
+
+	return nil
+}
+
+func reportSwitchportMetrics(reporter mb.ReporterV2, organizationID string, devices map[Serial]*Device) {
+	metrics := []mapstr.M{}
+	for _, device := range devices {
+		for _, switchport := range device.switchports {
+			metric := deviceDetailsToMapstr(device.details)
+
+			if switchport.port != nil {
+				metric["switch.port.id"] = switchport.port.PortID
```

Comment:
Should we have these fields as switchport.* rather than switch.port.* ?

Reply:
these are referred to a 'switch ports' in the [meraki docs](https://documentation.meraki.com/MS/Port_and_VLAN_Configuration/Switch_Ports), i think it's fine this way.